code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -454,6 +454,14 @@
 	var/our_pressure = air_contents.return_pressure()
 	var/our_temperature = air_contents.return_temperature()
 
+	if(our_pressure > pressure_limit + TANK_FRAGMENT_PRESSURE)
+		var/range = (our_pressure - (pressure_limit + TANK_FRAGMENT_PRESSURE)) / TANK_FRAGMENT_SCALE
+		var/turf/epicenter = get_turf(loc)
+		explosion(epicenter, round(range*0.25), round(range*0.5), round(range), round(range*1.5))
+		obj_break()
```

Comment:
maybe break before exploding, to avoid weird interactions of trying to break a qdeleted object?

Reply:
i'll change it yes

Reply:
Do test it after the change.